tools/ci_build/build.py

```diff
@@ -2267,7 +2267,7 @@ def main():
     if args.use_migraphx:
         args.use_rocm = True
 
-    if args.build_wheel or args.gen_doc or args.use_tvm:
+    if args.build_wheel or args.gen_doc or args.use_tvm or args.enable_training:
```

Comment:
if we don't need build_wheel, then why we still need enable pybind for enable_training?

Reply:
enable_training implies use of ortmodule (along with all other training modules).
Building ortmodule would require Building the python layer.
Let me know if this is too restrictive. I can disable this.
But typically I don't think we would ever need the entire training package without python.

Reply:
oh, okay, so how about the minimal training build for on-device training apis? that is not controlled by enbale_training any more?

Reply:
ok, I see there is a seperate enable_training_apis for that. OK, then this should be fine.

Reply:
But training also has nuget packages. Do it mean c# must be used with python?

Reply:
nuget packages for GPU training, or for on-device training? I assume we only target the nuget for the later one, which is controlled by enable_training_apis  flag. Is that right? @askhade  @baijumeswani

Reply:
The training nuget package is for on device training. It should be built with `--enable_training_apis` flag.

The `--enable_training` flag represents building all the training components (including ORTModule which cannot work without python wheel). So, if the user requests for all the training components, we enable python by default.

For `--enable_training_apis`, we do not include python by default.

Reply:
> nuget packages for GPU training, or for on-device training? I assume we only target the nuget for the later one, which is controlled by enable_training_apis flag. Is that right? @askhade @baijumeswani

nuget package is for both gpu as well as cpu (for on device training only). It is controlled by enable_training_apis flag.